name: Test Retry Failed Job
on:
  push:
    branches:
      - test-api-2

jobs:
  intentionally-fail:
    runs-on: ubuntu-latest
    steps:
      - name: Intentionally Fail Job
        run: exit 1
        id: fail_job

  rerun-failed-job:
    needs: intentionally-fail
    runs-on: ubuntu-latest
    if: always()
    steps:
      - name: Get Repo and Job ID
        run: |
          repo=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)
          job_id=${{ steps.fail_job.id }}
          echo "Repository: $repo"
          echo "Failed Job ID: $job_id"
        id: get_repo_job_id

      - name: Rerun Failed Job
        run: |
          token=$GITHUB_TOKEN
          repo=$(echo $GITHUB_REPOSITORY | cut -d '/' -f 2)
          job_id=${{ steps.get_repo_job_id.outputs.job_id }}
          url="https://api.github.com/repos/$repo/actions/runs/$job_id/rerun"
          response=$(curl -X POST -H "Authorization: token $token" $url)
          if [ $response -eq 201 ]; then
            echo "作业重新运行成功"
          else
            echo "作业重新运行失败"